### PR TITLE
[IT-3559] Update to use new snowflake access template

### PR DESCRIPTION
**Pull request by zaro0508**
switch to jinja template to accept multiple bucket Arn and Prefix

depends on https://github.com/Sage-Bionetworks/aws-infra/pull/413

### CODE REVIEW

**Comment by zaro0508** [comment]
This will generate the following template..
```
Description: >
  Setup cross account IAM access to a bucket/prefix for Snowflake.
  See https://docs.snowflake.com/en/user-guide/data-load-s3-config-storage-integration
AWSTemplateFormatVersion: 2010-09-09
Parameters:
  SnowflakeAccountArn:
    Type: String
  SnowflakeAccountExternalId:
    Type: String
Resources:
  SnowflakeServicePolicy:
    Type: 'AWS::IAM::ManagedPolicy'
    Properties:
      PolicyDocument:
        Version: '2012-10-17'
        Statement:

          - Effect: Allow
            Action:
              - 's3:GetObject'
              - 's3:GetObjectVersion'
            Resource:
              - "arn:aws:s3:::dev.datawarehouse.sagebase.org/warehouse/*"
          - Effect: Allow
            Action:
              - 's3:ListBucket'
              - 's3:GetBucketLocation'
            Resource: "arn:aws:s3:::dev.datawarehouse.sagebase.org"
            Condition:
              StringLike:
                's3:prefix': "warehouse/*"

  SnowflakeServiceRole:
    Type: "AWS::IAM::Role"
    Properties:
      Path: "/"
      ManagedPolicyArns:
        - !Ref SnowflakeServicePolicy
      AssumeRolePolicyDocument:
        Version: 2012-10-17
        Statement:
          - Effect: Allow
            Principal:
              AWS: !Sub ${SnowflakeAccountArn}
            Action:
              - sts:AssumeRole
              - sts:TagSession
            Condition:
              StringEquals:
                sts:ExternalId: !Sub ${SnowflakeAccountExternalId}
Outputs:
  SnowflakeServiceRoleArn:
    Value: !GetAtt SnowflakeServiceRole.Arn
    Export:
      Name: !Sub '${AWS::StackName}-SnowflakeServiceRoleArn'
```

File: sceptre/synapsedev/config/prod/snowflake-access.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-  url: https://{{stack_group_config.admincentral_cf_bucket}}.s3.amazonaws.com/aws-infra/v0.8.2/IAM/snowflake-bucket-access.yaml
+  url: https://{{stack_group_config.admincentral_cf_bucket}}.s3.amazonaws.com/aws-infra/v0.8.9/IAM/snowflake-bucket-access.j2
 stack_name: snowflake-access
 parameters:
-  BucketArn: arn:aws:s3:::dev.datawarehouse.sagebase.org
-  BucketPrefix: warehouse
   SnowflakeAccountArn: arn:aws:iam::365909334157:user/m2nb0000-s
   SnowflakeAccountExternalId: !ssm /infra/SnowflakeAccountExternalId
+sceptre_user_data:
+  Buckets:
```
Comment (thomasyu888) [inline sceptre/synapsedev/config/prod/snowflake-access.yaml:9]: Is this for an existing connection? Each s3 bucket needs separate policies.

Comment (zaro0508) [reply sceptre/synapsedev/config/prod/snowflake-access.yaml:9]: yes, existing connection.  just add another bucket to the `Buckets` list when setting up new connections.

**Review by marcomarasca — APPROVED** [review]

File: sceptre/synapsedev/config/prod/snowflake-access.yaml (hunk 1 above)
Comment (thomasyu888) [reply sceptre/synapsedev/config/prod/snowflake-access.yaml:9]: Thanks, I know there is also a production bucket somewhere, will that be updated in a separate PR?

Comment (zaro0508) [reply sceptre/synapsedev/config/prod/snowflake-access.yaml:9]: yes, add another bucket to the `Buckets` list in a follow on PR.

**Pull request merged**